src/hooks/usePublisher.ts

```diff
@@ -49,7 +64,7 @@ export default function usePublisher(roomId: RoomId | undefined) {
     }
 
     if (!hotClient) {
-        if (canMainPublish) {
+        if (canMainPublish || isTokenGated) {
```

Comment:
By design in token gated room main account cannot grant. That's why we gotta give the user a button that triggers the retrieving of the hot wallet.